expression/integration_test.go

```diff
@@ -2036,6 +2036,14 @@ func (s *testIntegrationSuite) TestTimeBuiltin(c *C) {
 	result.Check(testkit.Rows("<nil>"))
 	result = tk.MustQuery(`select tidb_parse_tso(-1)`)
 	result.Check(testkit.Rows("<nil>"))
+
+	// fix issue 10308
+	result = tk.MustQuery("select time(\"- -\");")
+	result.Check(testkit.Rows("00:00:00"))
```

Comment:
would mysql through a warning?

Reply:
> would mysql through a warning?

Both MySQL and TiDB through a warning:
```
mysql> show warnings;
+---------+------+---------------------------------------+
| Level   | Code | Message                               |
+---------+------+---------------------------------------+
| Warning | 1292 | Truncated incorrect time value: '- -' |
+---------+------+---------------------------------------+
1 row in set (0.00 sec)
```

Reply:
Got it. Could you also test the warning message?

Reply:
> Got it. Could you also test the warning message?

Done